### PR TITLE
I13 add expense

### PR DESCRIPTION
**Pull request by 1mozolacal**
## Related Issue
Added expense functionality expect merchant, Don't delete branch

  
  - [issue name](https://github.com/ps-toronto-team-4/.github/issues/ISSUENUMBER)

## Checks
- [X] My pull request adheres to the code style of this project
- [ ] My code requires changes to documentation
- [ ] I have updated the documentation as required
- [X] All the tests have passed

### CODE REVIEW

File: src/main/java/com/sapient/controller/ExpenseController.java [hunk 1]
```diff
@@ -32,9 +34,18 @@ public Record expense(@Argument String passwordHash, @Argument Integer id) {
     // Properties of a returned object from a @SchemaMapping method map to graphql fields of the same name.
     // Eg: "exceptionName".
     @MutationMapping
-    public Record createExpense(@Argument String passwordHash, @Argument String title, @Argument String description, @Argument Double amount) {
+    public Record createExpense(@Argument String passwordHash, @Argument String title, @Argument String description, @Argument Double amount, @Argument Date date, @Argument Integer categoryId, @Argument Integer merchantId, @Argument Integer recurrenceId) {
```
Comment (HKanwal) [inline src/main/java/com/sapient/controller/ExpenseController.java:37]: Line too long. Let's follow the intellij character limit.

File: src/main/java/com/sapient/controller/ExpenseController.java [hunk 2]
```diff
@@ -32,9 +34,18 @@ public Record expense(@Argument String passwordHash, @Argument Integer id) {
...
+        try {
+            return new ExpenseSuccess(expenseService.createExpense(passwordHash,title,description,amount,date,categoryId,merchantId,recurrenceId));
+        } catch (Exception e) {
+            return new FailurePayload(e.getClass().getSimpleName(), e.getMessage());
+        }
+    }
+
+    @MutationMapping
+    public Record updateExpense(@Argument String passwordHash, @Argument Integer id, @Argument String title, @Argument String description, @Argument Double amount, @Argument Date date, @Argument Integer categoryId, @Argument Integer merchantId, @Argument Integer recurrenceId){
```
Comment (HKanwal) [inline src/main/java/com/sapient/controller/ExpenseController.java:46]: Line too long here too.`

File: src/main/java/com/sapient/model/service/CategoryService.java [hunk 3]
```diff
@@ -39,7 +39,7 @@ public void deleteCategory(String passwordHash, Integer id) throws CategoryNotFo
         categoryDao.delete(category);
     }
 
-    public Category getCategory(String passwordHash, Integer id) throws CategoryNotFoundException, NotAuthorizedException {
+    public Category     getCategory(String passwordHash, Integer id) throws CategoryNotFoundException, NotAuthorizedException {
```
Comment (HKanwal) [inline src/main/java/com/sapient/model/service/CategoryService.java:42]: Indentation issue.

File: src/main/java/com/sapient/model/service/ExpenseService.java [hunk 4]
```diff
@@ -20,23 +22,62 @@ public class ExpenseService {
     @Autowired
     private UserService userService;
 
-    public Expense createExpense(String passwordHash, String title, String description, Double amount) throws NotAuthorizedException {
+	@Autowired
+	private CategoryService categoryService;
```
Comment (HKanwal) [inline src/main/java/com/sapient/model/service/ExpenseService.java:26]: Indentation issue.

**Review by jere-mie — CHANGES_REQUESTED** [review]
see requested changes 🤠

File: src/main/java/com/sapient/model/service/ExpenseService.java [hunk 5]
```diff
@@ -20,23 +22,62 @@ public class ExpenseService {
...
     	expense.setTitle(title);
     	expense.setAmount(amount);
     	expense.setDescription(description);
+		expense.setCategory(foundCategory);
+		//TODO merchant
     	expenseDao.save(expense);
     	return expense;
     }
 
+	public Expense updateExpense(String passwordHash, Integer id, String title, String description, Double amount, Date date, Integer categoryId, Integer merchantId, Integer recurrenceId) throws NotAuthorizedException, RecordNotFoundException {
```
Comment (HKanwal) [inline src/main/java/com/sapient/model/service/ExpenseService.java:53]: Line too long.

**Review by HKanwal — CHANGES_REQUESTED** [review]

**Pull request merged**